データベース設計

tech-o-zukky opened the pull request:
最終課題：データベース設計を行いました。
レビューをお願いします。

# 対象ファイル
README.md

# ER図
[Gyazo URL](https://gyazo.com/51e64bba638379dff390724768c71d93)

tech-o-zukky committed:
DB設計：Readme記入

tech-o-zukky committed:
DB設計：Readme記入修正

TakumaKawashima commented on README.md:
```diff
@@ -1,24 +1,67 @@
+| --------------- | ------ | ----------- |
+| nickname        | string | null: false |
+| email           | string | null: false |
+| password        | string | null: false |
```
こちらのカラムはdeviseのgemを使用しての実装になるため、deviseが予め用意している
encrypted_passwordというカラムを用意しておきましょう。

TakumaKawashima commented on README.md:
```diff
@@ -1,24 +1,67 @@
+| Column                      | Type        | Options                         |
+| --------------------------- | ----------- | ------------------------------- |
+| item_name                   | string      | null: false                     |
+| item_explanation            | string      | null: false                     |
```
データ型は「text型」に変更しておきましょう。

TakumaKawashima commented on README.md:
```diff
@@ -1,24 +1,67 @@
+# テーブル設計
 
-Things you may want to cover:
+## members テーブル（ユーザー管理テーブル）
```
membersよりかは、usersテーブルにしておきましょう。

現状ですと、devise導入の際に命名をmembersで指定する必要があります。

TakumaKawashima commented on README.md:
```diff
@@ -1,24 +1,67 @@
+| item_scheduled_delivery_id  | integer     | null: false                     |
+| sell_price                  | integer     | null: false                     |
+| member_id                   | references  | null: false, foreign_key: true  |
+| purchase_id                 | references  | foreign_key: true               |
```
```
| purchase_id                 | references  | foreign_key: true               |
```
こちらは不要ですので、削除しておきましょう。

TakumaKawashima commented on README.md:
```diff
@@ -1,24 +1,67 @@
+| ------------------------ | ----------- | ------------------------------- |
+| zipcode                  | string      | null: false                     |
+| prefecture_id            | integer     | null: false                     |
+| address1                 | string      | null: false                     |
+| address2                 | string      | null: false                     |
```
```
| address1                 | string      | null: false                     |
| address2                 | string      | null: false                     |
```
address1とaddress2ですと、他者から見た場合、中身が想像しにくいため
cityとaddressなどにしておきましょう。

TakumaKawashima commented on README.md:
```diff
@@ -1,24 +1,67 @@
+| address2                 | string      | null: false                     |
+| address_building_name    | string      |                                 |
+| telephone                | string      | null: false                     |
+| item_id                  | references  | null: false, foreign_key: true  |
+| member_id                | references  | null: false, foreign_key: true  |
```
```
| item_id                  | references  | null: false, foreign_key: true  |
| member_id                | references  | null: false, foreign_key: true  |
```
購入テーブルには、上記２つのカラムだけで問題ありません。

その他のaddressカラムなどは、addressテーブル（住所登録）のテーブルを作成しましょう。

TakumaKawashima reviewed (commented):
レビュー依頼ありがとうございます。
コメントいたしましたので、ご確認をお願いいたします。

tech-o-zukky committed:
DB設計：指摘を受け再修正

tech-o-zukky commented:
ありがとうございます。
上記の指摘をうけ、修正しました。
再レビューよろしくお願いします。

[ER図](https://gyazo.com/f31a65efef5bd9d8c46d8141bdd11c3d)

s-aka-moto commented on README.md:
```diff
@@ -1,24 +1,76 @@
-* Database initialization
+| Column                      | Type        | Options                         |
+| --------------------------- | ----------- | ------------------------------- |
+| item_name                   | text        | null: false                     |
```
商品名が255字を超えることはないかと存じますので、string型で実装しましょう！
`| item_name                   | text        | null: false                     |`

s-aka-moto commented on README.md:
```diff
@@ -1,24 +1,76 @@
+| item_prefecture_id          | integer     | null: false                     |
+| item_scheduled_delivery_id  | integer     | null: false                     |
+| sell_price                  | integer     | null: false                     |
+| user_id                     | references  | null: false, foreign_key: true  |
```
reference型は_idが不要です！
| user`_id `                    | references  | null: false, foreign_key: true  |

s-aka-moto commented on README.md:
```diff
@@ -1,24 +1,76 @@
+
+| Column                   | Type        | Options                         |
+| ------------------------ | ----------- | ------------------------------- |
+| item_id                  | references  | null: false, foreign_key: true  |
+| user_id                  | references  | null: false, foreign_key: true  |
```
references型は_idが不要です！
```
| item_id                  | references  | null: false, foreign_key: true  |
| user_id                  | references  | null: false, foreign_key: true  |
```

s-aka-moto commented on README.md:
```diff
@@ -1,24 +1,76 @@
+| address                  | string      | null: false                     |
+| address_building_name    | string      |                                 |
+| telephone                | string      | null: false                     |
+| purchase_id              | references  | null: false, foreign_key: true  |
```
こちらも同様です！
| purchase`_id`              | references  | null: false, foreign_key: true  |

s-aka-moto commented on README.md:
```diff
@@ -1,24 +1,76 @@
+| item_category_id            | integer     | null: false                     |
+| item_status_id              | integer     | null: false                     |
+| item_shipping_fee_status_id | integer     | null: false                     |
+| item_prefecture_id          | integer     | null: false                     |
```
都道府県の情報は購入時にも使用するため、こちらもprefecture_idとすると良いですね！

s-aka-moto reviewed (commented):
レビュー依頼ありがとうございます！
コメントしましたので、ご確認をお願いいたします！

tech-o-zukky committed:
DB設計：指摘事項修正

・items.item_nameをstring型に修正
・references型のカラム名から「_id」を削除
・itemsのカラム名修正（item_prefecture_id→prefecture_id）

tech-o-zukky committed:
Revert "DB設計：指摘事項修正"

This reverts commit b74d56c26d96564551efcc59a2c0005e33be41fc.

tech-o-zukky committed:
DB設計：指摘事項修正

・items.item_nameをstring型に修正
・references型のカラム名から「_id」を削除
・itemsのカラム名修正（item_prefecture_id→prefecture_id）

tech-o-zukky committed:
Revert "DB設計：指摘事項修正"

This reverts commit 9dcba469b2a0c17818d7207ddc5502cab784f54f.

tech-o-zukky committed:
DB設計：指摘事項修正

・items.item_nameをstring型に修正
・itemsのカラム名修正（item_prefecture_id→prefecture_id）
・references型のカラム名から「_id」を削除

tech-o-zukky commented:
ありがとうございます。
上記の指摘をうけ、再度修正しました。
再レビューよろしくお願いします。
[ER図（修正版）](https://gyazo.com/b000d8bfdf7e191f2e05b2fcfc507024)

# 修正点
- items.item_nameをstring型に修正
- itemsのカラム名修正（item_prefecture_id→prefecture_id）
- references型のカラム名から「_id」を削除

chirikuzu reviewed (commented):
レビュー依頼ありがとうございます。
LGTMとさせていただきます。実装お疲れ様でした。
マージ、プルを実行後、次の実装に進まれてください。
引き続き実装頑張っていきましょう。
